functions/Remove-DbaBackup.ps1

```diff
@@ -0,0 +1,225 @@
+                throw "RetentionPeriod '$UserFriendlyRetention' format invalid! See Get-Help for correct formatting and examples."
+            }
+
+            switch ($Units)
+            {
+                'h' { $UnitString = 'Hours'; [datetime]$ReturnDatetime = (Get-Date).AddHours(-$Value)  }
+                'd' { $UnitString = 'Days';  [datetime]$ReturnDatetime = (Get-Date).AddDays(-$Value)   }
+                'w' { $UnitString = 'Weeks'; [datetime]$ReturnDatetime = (Get-Date).AddDays(-$Value*7) }
+                'm' { $UnitString = 'Months';[datetime]$ReturnDatetime = (Get-Date).AddMonths(-$Value) }
+            }
+            Write-Verbose "Retention set to '$Value' $UnitString. Retention date/time '$ReturnDatetime'"
+            $ReturnDatetime
+        }
+
+        # Initialize stuff
+        $Start = Get-Date
+	}
+	PROCESS
+	{
+		# Process stuff
+        Write-Output ("Started at $Start")
+        Write-Output ("Removing backups from '$BackupFolder'")
+
+        # Convert Retention Value to an actual DateTime
+        try {
+            $RetentionDate = Convert-UserFriendlyRetentionToDatetime -UserFriendlyRetention $RetentionPeriod
+            Write-Output "Backup Retention Date set to '$RetentionDate'"
+        } catch {
+            throw $_
+        }
+
+        # Generate list of files that are to be removed
+        $FilesToDelete = Get-ChildItem "$BackupFolder" -Filter "*.$BackupFileExtenstion" -Recurse | `
+            Where-Object {$_.LastWriteTime -lt $RetentionDate}
+            
+        # Filter out unarchived files if -CheckArchiveBit parameter is used
+        if ($CheckArchiveBit.IsPresent) {
+            Write-Output 'Removing only archived files'
+            $FilesToDelete = $FilesToDelete | Where-Object {$_.attributes -notmatch "Archive"} 
+        }
+        
+        # Perform the deletion or show which file will be deleted if WhatIf is used
+        foreach ($file in $filestodelete) { 
+            If ($Pscmdlet.ShouldProcess($env:computername, "Deleting $($file.fullname)'")) {
+                try {
+                    $file.FullPath | Remove-Item -Force -Verbose 4>&1
+                } catch {
+                    throw $_
+                }
+            }
+        }
+ 
+        # Remove empty backup folders if RemoveEmptyBackupFolders is passed in
+        if ($RemoveEmptyBackupFolders.IsPresent -and $Pscmdlet.ShouldProcess($env:computername, "Removing empty folders under '$BackupFolder\*'")) {
+            try {
+                # Keep looping until we dont find any more
+                while (Get-ChildItem -Path $BackupFolder -Recurse | Where-Object {$_.PSIsContainer -eq $true `
+                    -and (Get-ChildItem -Path $_.FullName) -eq $null}) {
+                    Get-ChildItem -Path $BackupFolder -Recurse | Where-Object {$_.PSIsContainer -eq $true `
+                        -and (Get-ChildItem -Path $_.FullName) -eq $null} | Remove-Item -Force -Verbose 4>&1
```

Comment:
Adding -Verbose here removes the ability for the user to choose verbose output, it will always display this regardless of whether people run your command with or without -Verbose. Consider removing.